packages/opentelemetry-types/src/trace/instrumentation/Plugin.ts

```diff
@@ -86,6 +82,76 @@ export interface PluginConfig {
   enhancedDatabaseReporting?: boolean;
 }
 
+/**
+ * These options are used by dns module plugins.
+ */
+export interface DNSPluginOptions {
+  /**
+   * Used by dns plugin. Ignores tracing for host names which match one of
+   * the configured matchers by either being an exact string match, matching
+   * a regular expression, or evaluating to true.
+   */
+  ignoreHostnames?: IgnoreMatcher[];
+}
+
+export interface HttpCustomAttributeFunction {
+  (
+    span: Span,
+    request: ClientRequest | IncomingMessage,
+    response: IncomingMessage | ServerResponse
+  ): void;
+}
+
+/**
+ * These options are used by http plugins like http, https, and http2.
+ */
+export interface HttpPluginOptions {
+  /** Not trace all incoming requests that match paths */
+  ignoreIncomingPaths?: IgnoreMatcher[];
+  /** Not trace all outgoing requests that match urls */
+  ignoreOutgoingUrls?: IgnoreMatcher[];
+  /** Function for adding custom attributes */
+  applyCustomAttributesOnSpan?: HttpCustomAttributeFunction;
+  /** The primary server name of the matched virtual host. */
+  serverName?: string;
+}
+
+export type PropagateTraceHeaderCorsUrl = string | RegExp;
+
+export type PropagateTraceHeaderCorsUrls =
+  | PropagateTraceHeaderCorsUrl
+  | PropagateTraceHeaderCorsUrl[];
+
+/**
+ * XMLHttpRequest config
+ */
+export interface XMLHttpRequestPluginOptions {
+  // the number of timing resources is limited, after the limit
+  // (chrome 250, safari 150) the information is not collected anymore
+  // the only way to prevent that is to regularly clean the resources
+  // whenever it is possible, this is needed only when PerformanceObserver
+  // is not available
+  clearTimingResources?: boolean;
+  // urls which should include trace headers when origin doesn't match
+  propagateTraceHeaderCorsUrls?: PropagateTraceHeaderCorsUrls;
+}
+
+/**
+ * This is a configuration section where plugin authors
+ * can define their own custom configuration options.
+ */
+export interface CustomPluginOptions {
+  [key: string]: number | string | boolean | undefined;
+}
+
+export interface PluginOptions {
+  database?: DatabasePluginOptions;
+  dns?: DNSPluginOptions;
+  http?: HttpPluginOptions;
+  xhr?: XMLHttpRequestPluginOptions;
+  custom?: CustomPluginOptions;
```

Comment:
what about: 
`[custom: string]: any;`
People would be able to add their own plugin like this:
```ts
const options: PluginOptions = {
  http: {
    serverName: 'hello',
  },
  amqp: {
    something: 'else'
  }
}
```
I don't have a strong opinion on this, I'm just curious.

Reply:
There is a whole section named `custom` which allows you to put custom config options there.